src/shell/shell_cmds.cpp

```diff
@@ -2242,3 +2243,290 @@ void DOS_Shell::CMD_VOL(char* args)
 	const auto trimmed_label = trim(label);
 	WriteOut(MSG_Get("SHELL_CMD_VOL_OUTPUT"), drive, trimmed_label, high, low);
 }
+
+static std::vector<std::string> cmd_move_parse_sources(const char*& args)
+{
+	bool parsing             = true;
+	bool inside_quote        = false;
+	bool stop_next_letter    = false;
+	bool continue_after_stop = false;
+	std::string word;
+	std::vector<std::string> sources;
+
+	// Sources are seperated by commas and terminated by a trailing space
+	while (parsing) {
+		switch (*args) {
+		case '"':
+			inside_quote = !inside_quote;
+			args++;
+			break;
+		case ' ':
+			if (inside_quote) {
+				word.push_back(*args++);
+			} else {
+				if (!word.empty()) {
+					sources.emplace_back(std::move(word));
+					word             = {};
+					stop_next_letter = true;
+					continue_after_stop = false;
+				}
+				args++;
+			}
+			break;
+		case ',':
+			stop_next_letter = true;
+			continue_after_stop = true;
+			if (!word.empty()) {
+				sources.emplace_back(std::move(word));
+				word = {};
+			}
+			args++;
+			break;
+		case 0:
+			if (!word.empty()) {
+				sources.emplace_back(std::move(word));
+			}
+			parsing = false;
+			break;
+		default:
+			if (stop_next_letter) {
+				parsing             = continue_after_stop;
+				continue_after_stop = false;
+				stop_next_letter    = false;
+			} else {
+				word.push_back(*args++);
+			}
+		}
+	}
+	return sources;
+}
+
+static std::string cmd_move_parse_destination(const char*& args)
+{
+	std::string destination;
+	bool parsing      = true;
+	bool inside_quote = false;
+
+	while (parsing) {
+		switch (*args) {
+		case '"':
+			inside_quote = !inside_quote;
+			args++;
+			break;
+		case ' ':
+			if (inside_quote) {
+				destination.push_back(*args++);
+			} else {
+				args++;
+				parsing = false;
+			}
+			break;
+		case '\0': parsing = false; break;
+		default: destination.push_back(*args++);
+		}
+	}
+	return destination;
+}
+
+struct cmd_move_arguments {
+	std::vector<std::string> sources = {};
+	std::string destination          = {};
+	const char* error                = nullptr;
+};
+
+cmd_move_arguments cmd_move_parse_arguments(const char* args)
+{
+	cmd_move_arguments ret;
+	ret.sources = cmd_move_parse_sources(args);
+	if (ret.sources.empty()) {
+		ret.error = "SHELL_MISSING_PARAMETER";
+		return ret;
+	}
+	ret.destination = cmd_move_parse_destination(args);
+	if (ret.destination.empty()) {
+		ret.error = "SHELL_MISSING_PARAMETER";
+		return ret;
+	}
+	// Check for extra arguments
+	while (*args) {
+		if (*args++ != ' ') {
```

Comment:
At least we should document that the parse methods have a side-effect of mutating the passed in char pointer, and subsequent parsing relies on this behaviour. At the very least I'd still not destroy the args pointer passed to the main parse function.

Reply:
Just to be clear, it the actual string behind the `char *` never gets mutated.  The main function takes a `const char *`.  The 2 helper functions take `const char*&` so that they can advance the pointer past what's already been parsed.

I can certainly add a comment stating that.  Or if you have a better idea in mind I'm all ears.

Reply:
> Just to be clear, it the actual string behind the `char *` never gets mutated. The main function takes a `const char *`. The 2 helper functions take `const char*&` so that they can advance the pointer past what's already been parsed.
> 
> I can certainly add a comment stating that. Or if you have a better idea in mind I'm all ears.

Yeah I get that; it's a mutable pointer to a const char string. In general, I dislike mutating passed in arguments. This might seem a petty thing in this case, and you can figure out what's happening in a minute or two, so I'm not asking you to change the whole thing, just document that the passed-in args pointer get advanced, and further parse functions rely on that behavoiur.

The cleanest solution would be to pass back a `string_view` of the remaining part. If this was a complicated parses with tens of parse functions, I'd definitely say do that. But it would be an overkill in this case, so just document it please.

Reply:
Documented